src/app/search/SearchFilterInput.tsx

```diff
@@ -62,165 +64,87 @@ const filterNames = [
  * A reusable, autocompleting item search input. This is an uncontrolled input that
  * announces its query has changed only after some delay.
  */
-export default class SearchFilterInput extends React.Component<Props, State> {
-  state: State = { liveQuery: '', filterHelpOpen: false };
-  private textcomplete: Textcomplete;
-  private inputElement = React.createRef<HTMLInputElement>();
-  private debouncedUpdateQuery = _.debounce(this.props.onQueryChanged, 500);
-
-  componentWillUnmount() {
-    if (this.textcomplete) {
-      this.textcomplete.destroy();
-      this.textcomplete = null;
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.searchConfig !== this.props.searchConfig) {
-      this.setupTextcomplete();
-    }
-    if (
-      prevProps.searchQueryVersion !== this.props.searchQueryVersion &&
-      this.props.searchQuery !== undefined
-    ) {
-      this.setState({ liveQuery: this.props.searchQuery });
-    }
-  }
-
-  render() {
-    const { alwaysShowClearButton, placeholder, children, autoFocus } = this.props;
-    const { liveQuery, filterHelpOpen } = this.state;
+export default React.forwardRef(function SearchFilterInput(
+  {
+    searchQueryVersion,
+    searchConfig,
+    searchQuery,
+    alwaysShowClearButton,
+    placeholder,
+    children,
+    autoFocus,
+    onQueryChanged,
+    onClear,
+  }: Props,
+  ref: React.Ref<{ focusFilterInput(): void; clearFilter(): void }>
```

Comment:
I pulled the ref type out into
```
export type SearchFilterInputRefType = { focusFilterInput(): void; clearFilter(): void };
```
and used that instead of `SearchFilterInput` when creating refs and the compiler was happy, not entirely convinced if that's actually a bandaid or a proper fix yet though.

Reply:
Yup, I did too - but my internet went down before I could push the fix.